libs/shared/v2/src/lib/models/asset-search.ts

```diff
@@ -33,22 +36,94 @@ export interface AssetSearchResult {
   data: AssetSearchResultItem[];
 }
 
-export const makeEmptyAssetSearchResults = (): AssetSearchResult => ({
-  page: {
-    size: 0,
-    offset: 0,
-    total: 0,
-  },
-  data: [],
-});
+export class AssetSearchResultItemStudyDTO implements AssetSearchResultItemStudy {
+  @Expose()
+  @IsString()
+  studyId!: string;
+
+  @Expose()
+  @IsString()
+  geomText!: string;
+}
+
+export class AssetSearchResultItemContactDTO implements AssetSearchResultItemContact {
+  @Expose()
+  @IsIn(AssetContactRoles)
+  role!: AssetContactRole;
+
+  @Expose()
+  @IsNumber()
+  contactId!: number;
+}
+
+export class AssetSearchResultItemUseDTO implements AssetSearchResultItemUse {
+  @Expose()
+  @IsBoolean()
+  isAvailable!: boolean;
+}
+
+export class AssetSearchResultItemDTO implements AssetSearchResultItem {
+  @Expose()
+  @IsNumber()
+  assetId!: number;
+
+  @Expose()
+  @IsNumber()
+  createDate!: number & DateIdBrand;
+
+  @Expose()
+  @IsString()
+  titlePublic!: string;
+
+  @Expose()
+  @IsArray()
+  @Type(() => AssetSearchResultItemStudyDTO)
+  studies!: AssetSearchResultItemStudyDTO[];
+
+  @Expose()
+  @IsString()
+  assetKindItemCode!: string;
+
+  @Expose()
+  @IsArray()
+  @Type(() => AssetSearchResultItemContactDTO)
+  assetContacts!: AssetSearchResultItemContactDTO[];
+
+  @Expose()
+  @IsString()
+  assetFormatItemCode!: string;
+
+  @Expose()
+  @IsArray()
+  manCatLabelRefs!: string[];
+
+  @Expose()
+  @Type(() => AssetSearchResultItemUseDTO)
+  internalUse!: AssetSearchResultItemUseDTO;
+  @Expose()
+  @Type(() => AssetSearchResultItemUseDTO)
+  publicUse!: AssetSearchResultItemUse;
+}
 
 export class AssetSearchResultDTO implements AssetSearchResult {
+  @Expose()
   page!: PageStats;
-  data!: AssetSearchResultItem[];
+  @Expose()
+  @Type(() => AssetSearchResultItemDTO)
+  data!: AssetSearchResultItemDTO[];
```

Comment:
I personally like to type DTO fields by interface rather than by concrete class - IMO the DTO should really only be used for parsing, and the interface everywhere else.
```suggestion
  data!: AssetSearchResultItem[];
```